src/components/edit/event/EventDialog.js

```diff
@@ -142,8 +142,7 @@ export class EventDialog extends Component {
             setPeriod({
                 id: defaultPeriod,
             });
-        } else {
-            // Fallback to default start/end dates
+        } else if (!startDate && !endDate) {
```

Comment:
Is the "else" case handled correctly? (if there is a plausible "else" case)

Reply:
It is no plausible else case here - it means that a period is already set - and we should not apply a default.